### DISCUSSION TITLE
Aim-based experiment tracker for spaCy

### DISCUSSION THREAD

**Discussion by SGevorg** [explosion/spaCy#10832] [Show & Tell]
Hi all, excited to share with you aim-spaCy! 🎉 

It's an Aim plugin for spaCy that automatically logs the spaCy metrics and displaCy visualizations as images.
Both on training and evaluation.

Aim has a powerful Images Explorer that allows to compare 1000s of tracked images in-depth.
aim-spaCy will allow to seamlessly compare 1000s of displaCy visualizations at once with a few clicks. 🙌 

### A use-case example
In the gif below, we have tracked 100+ displaCy Dependency Tree and NER visualizations for one spaCy inference run.
With 2 clicks we can compare two visualizations of the same sentence at each step:
- group by `record.step`
- stack by the last grouping param (the step)
- [Bonus] you can compare across 100s of runs too

**Basically this will allow to find where and why (thanks to displaCy magic) models misbehave / underperform.**

![Screen Recording for Gev (1)](https://user-images.githubusercontent.com/3179216/169555016-188638da-5715-4a2d-91dd-a14543513702.gif)

### Other perks
- Free, self-hosted and open-source
- Easy to embed in your jupyter notebook
- Compare 1000s of high-dimensional experiments (100s of metrics, images, text tracked per experiment)

web: [aimstack.io/spacy](https://bit.ly/3MqNV2L)
code: [github.com/aimhubio/aim-spacy](https://github.com/aimhubio/aim-spacy)

Would love your feedback on aim-spaCy. What else would you like us to add to the project?

**Comment by philipvollet** [comment]
Tested around and love it! Sent you a DM via Twitter with some questions, super curious about how the project continues to develop 💙
